libs/community/langchain_community/document_loaders/parsers/pdf.py

```diff
@@ -78,6 +203,192 @@ def extract_from_images_with_rapidocr(
     return text
 
 
+# Type to change the function to convert images to text.
+CONVERT_IMAGE_TO_TEXT = Optional[Callable[[Iterable[np.ndarray]], Iterator[str]]]
```

Comment:
MAJOR:

Why not use an ImageBlobParser w/ the regular Blob to Document interface. it'll allow reusing the image logic for images that do not originate from pdfs (e.g., to re-use for a web crawler)

---

A PDF parser doesn't would accept a parser as part of the initializer

```python
class PDFParser(...):
   def __Init__(self, ...  *, ..., image_blob_parser: Optional[BlobParser] = None):
      pass
```

If the image_pdf_parser is provided, then it'll be used for OCR purposes.

Reply:
I've done it! There are now 3 ImageBlogParsers!